test/unit/mdc-tabs/mdc-tab-bar-foundation.test.js

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import {setupFoundationTest} from '../helpers/setup';
+import {verifyDefaultAdapter, captureHandlers} from '../helpers/foundation';
+import {createMockRaf} from '../helpers/raf';
+
+import MDCTabBarFoundation from '../../../packages/mdc-tabs/tab-bar/foundation';
+
+suite('MDCTabBarFoundation');
+
+test('exports cssClasses', () => {
+  assert.isOk('cssClasses' in MDCTabBarFoundation);
+});
+
+test('default adapter returns a complete adapter implementation', () => {
+  verifyDefaultAdapter(MDCTabBarFoundation, [
+    'addClass', 'removeClass', 'bindOnMDCTabSelectedEvent',
+    'unbindOnMDCTabSelectedEvent', 'registerResizeHandler',
+    'deregisterResizeHandler', 'getOffsetWidth', 'setStyleForIndicator',
+    'getOffsetWidthForIndicator', 'notifyChange',
+    'getNumberOfTabs', 'isTabActiveAtIndex', 'setTabActiveAtIndex',
+    'isDefaultPreventedOnClickForTabAtIndex',
+    'setPreventDefaultOnClickForTabAtIndex', 'measureTabAtIndex',
+    'getComputedWidthForTabAtIndex', 'getComputedLeftForTabAtIndex',
+  ]);
+});
+
+function setupTest() {
+  const {foundation, mockAdapter} = setupFoundationTest(MDCTabBarFoundation);
+  const {UPGRADED} = MDCTabBarFoundation.cssClasses;
+  const {TAB_SELECTOR, INDICATOR_SELECTOR} = MDCTabBarFoundation.strings;
+  const tabHandlers = captureHandlers(mockAdapter, 'bindOnMDCTabSelectedEvent');
+
+  return {foundation, mockAdapter, UPGRADED, TAB_SELECTOR, INDICATOR_SELECTOR, tabHandlers};
+}
+
+test('#init adds upgraded class to tabs', () => {
+  const {foundation, mockAdapter, UPGRADED} = setupTest();
+
+  foundation.init();
+  td.verify(mockAdapter.addClass(UPGRADED));
+});
+
+test('#init registers listeners', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const {isA} = td.matchers;
+
+  foundation.init();
+  td.verify(mockAdapter.bindOnMDCTabSelectedEvent());
+  td.verify(mockAdapter.registerResizeHandler(isA(Function)));
+});
+
+test('#init sets active tab if index of active tab on init() is > 0', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const activeTabIndex = 2;
+  const numberOfTabs = 5;
+
+  td.when(mockAdapter.getNumberOfTabs()).thenReturn(numberOfTabs);
+  td.when(mockAdapter.isTabActiveAtIndex(activeTabIndex)).thenReturn(true);
+
+  foundation.init();
+
+  assert.isTrue(mockAdapter.isTabActiveAtIndex(activeTabIndex));
+});
+
+test('#init sets -webkit-transform and transform for indicator', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  foundation.init();
+  raf.flush();
+
+  td.verify(mockAdapter.setStyleForIndicator(td.matchers.anything(), td.matchers.anything()));
```

Comment:
Correct. I added two additional verifications as well for all of the styles `init()` sets on the indicator.